astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -8,14 +8,138 @@
+
+    propagate_uncertainties : `bool` or ``None``, optional
+        If ``None`` the result will have no uncertainty. If ``False`` the
+        result will have a copied version of the first operand (or the
+        second if the first had no uncertainty). In case this is ``True``
+        the result will have a correctly propagated uncertainty from the
+        uncertainties of the operands but this assumes that the uncertainties
+        are `StdDevUncertainty`. Default is ``True``.
+
+    handle_mask : callable, ``False`` or ``None``, optional
+        If ``None`` the result will have no mask. If ``False`` the result
+        will have a copied version of the mask of the first operand (or
+        if the first operand has no mask then the one from the second is
+        taken). If it is a callable then the specified callable must
+        create the results ``mask`` and if necessary provide a copy.
+        Default is `numpy.logical_or`.
+
+    handle_meta : callable, ``False`` or ``None``, optional
+        If ``None`` the result will have no meta. If ``False`` the result
+        will have a copied version of the meta of the first operand (or
+        if the first operand has no meta then the one from the second is
+        taken). If it is a callable then the specified callable must
+        create the results ``meta`` and if necessary provide a copy.
+        Default is ``False``.
+
+    compare_wcs : callable, ``False`` or ``None``, optional
+        If ``None`` the result will have no wcs and no comparison between
+        the wcs of the operands is made. If ``False`` the result will have
+        the wcs of the first operand (or second if the first had ``None``).
+        If it is a callable then the specified callable must
+        compare the ``wcs``. The resulting ``wcs`` will be like if ``False``
+        was given otherwise it raises a ``ValueError`` if the comparison was
+        not successful. Default is ``False``.
+
+    uncertainty_correlation : ``Number`` or `~numpy.ndarray`, optional
+        The correlation (rho) is defined between the uncertainties in
+        ``sigma_AB = sigma_A * sigma_B * rho`` (Latex?). If
+        ``propagate_uncertainties`` is *not* ``True`` this will be ignored.
+        A value of ``0`` means uncorrelated (which is also the default).
+        TODO: TeX of formula?
+
+    kwargs :
+        Any other parameter that should be passed to the callables used.
+
+    Returns
+    -------
+    result : `~astropy.nddata.NDData`-like
+        The resulting dataset
+
+    Notes
+    -----
+    In case a ``callable`` is used for ``mask``, ``wcs`` or ``meta`` the
+    callable must accept the corresponding attributes as first two
+    parameters. If the callable also needs additional parameters these can be
+    defined as ``kwargs`` and must start with ``"wcs_"`` (for wcs callable) or
+    ``"meta_"`` (for meta callable). This startstring is removed before the
+    callable is called.
```

Comment:
This works, but I'm wondering if there is a better way, e.g. one of the `kwargs` could be a `dict` with keys `wcs`, `meta`, etc, whose value is a dictionary of optional arguments for the callables. It seems a little magical for the leading string to be automatically stripped out.